src/microbe_stage/editor/CellEditorComponent.GUI.cs

```diff
@@ -174,9 +175,42 @@ private void UpdateHitpoints(float hp)
         hpLabel.Value = hp;
     }
 
-    private void UpdateStorage(float storage)
+    private void UpdateStorage(float nominalStorage, Dictionary<Compound, float> storage)
     {
-        storageLabel.Value = (float)Math.Round(storage, 1);
+        storageLabel.Value = nominalStorage;
+
+        var tooltip = ToolTipManager.Instance.GetToolTip("storageDetails", "editor");
+        var tooltipLoaded = tooltip != null;
+
+        if (storage.Count == 0)
+        {
+            storageLabel.UnRegisterFirstToolTipForControl();
+
+            return;
+        }
+
+        var description = new LocalizedStringBuilder(100);
+
+        bool first = true;
+
+        foreach (var entry in storage)
+        {
+            if (!first)
+                description.Append("\n");
+
+            first = false;
+
+            description.Append(entry.Key.Name);
+            description.Append(": ");
+            description.Append(entry.Value);
+        }
+
+        storageLabel.RegisterToolTipForControl("storageDetails", "editor");
+
+        if (tooltipLoaded)
+            tooltip!.Description = description.ToString();
+        else
+            GD.PrintErr("Can't update storage tooltip");
```

Comment:
The "RegisterToolTipForControl" already fails if tooltip is not loaded, I assume so you should just early exit and print an error on line 183 (instead of storing the info in a bool).